src/test/java/com/icthh/xm/uaa/security/DomainTokenServicesUnitTest.java

```diff
@@ -155,6 +180,40 @@ public void testRefreshToken() throws Exception {
         assertEquals(USER_KEY, ((Map<String, String>) auth.getDetails()).get(AUTH_USER_KEY));
     }
 
+    @Test
+    public void testRefreshTokenForDeactivatedUser() throws Exception {
+
+        when(userService.findOneByLogin(LOGIN)).thenAnswer(invocation -> {
+            User user = new User();
+            user.setActivated(false); // user is not active
+            user.setUserKey(USER_KEY);
+            return Optional.of(user);
+        });
+
+        OAuth2AccessToken accessToken = tokenServices.createAccessToken(createAuthentication(LOGIN, TENANT, ROLE));
+
+        assertNotNull(accessToken);
+        assertNotNull(accessToken.getRefreshToken());
+        assertNotNull(accessToken.getRefreshToken().getValue());
+
+        Map<String, String> params = new HashMap<>();
+        params.put("grant_type", "refresh_token");
+        params.put("refresh_token", accessToken.getRefreshToken().getValue());
+
+        TokenRequest tokenRequest = new TokenRequest(params, CLIENT, null, "refresh_token");
+        try {
+            tokenServices.refreshAccessToken(accessToken.getRefreshToken().getValue(), tokenRequest);
+            fail( "Expected UserDeniedAuthorizationException to be thrown" );
```

Comment:
it's small
better to user @Test(expect=Exception.class) like this

Reply:
But how I can execute my `verify` methods in that case? Move this single line to separate method with `@Test(expect=UserDeniedAuthorizationException.class)`?

Reply:
ok, how about
  @Rule
  public final ExpectedException exception = ExpectedException.none();
or write two tests?

Reply:
This was another option really, but I didn't like it, because we have to create a separate field for this and instantiate an `ExpectedException` object. What do you think?